ecs-init/docker/docker_test.go

```diff
@@ -751,3 +755,158 @@ func TestGetDockerSocketBind(t *testing.T) {
 		})
 	}
 }
+
+func TestStartAgentWithExecBinds(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	containerID := "container id"
+	isPathValid = func(path string, isDir bool) bool {
+		return true
+	}
+	expectedAgentBinds += 2
+	defer func() {
+		expectedAgentBinds = expectedAgentBindsUnspecifiedPlatform
+		isPathValid = defaultIsPathValid
+	}()
+
+	mockFS := NewMockfileSystem(mockCtrl)
+	mockDocker := NewMockdockerclient(mockCtrl)
+
+	mockFS.EXPECT().ReadFile(config.InstanceConfigFile()).Return(nil, errors.New("not found")).AnyTimes()
+	mockFS.EXPECT().ReadFile(config.AgentConfigFile()).Return(nil, errors.New("not found")).AnyTimes()
+	mockDocker.EXPECT().CreateContainer(gomock.Any()).Do(func(opts godocker.CreateContainerOptions) {
+		validateCommonCreateContainerOptions(opts, t)
```

Comment:
We're validating the expected number of bind mounts, which is good. But can we also verify in here that the actual contents are ok? Let's do this check after calling `validateCommonCreateContainerOptions`, as opposed to doing the checks inside `validateCommonCreateContainerOptions`